### PR TITLE
Projectile blocking fixes

### LINKED ISSUE #10969: Riot Shields are not blocking Syringes properly
Steps to reproduce:
Have shield
Get shot at
Message says you block the syringe
Syringe actually hit you, chems take effect

This wasn't the case previously so I don't know what could be causing this

### PR DESCRIPTION
**Pull request by phil235**
- Removes some unneeded code(flag = "magic" when already inheriting it from parent, adding a 0 argument to on_hit when it already defaults to 0).
- Fixes syringe dart not respecting shield blocking. Fixes #10969 
- Projectile doesn't show the hit message anymore if 100% blocked (shield blocking mostly) so you only get the block message. (but the hit is still logged)
- Fixes deflected dart projectiles not being logged.

### CODE REVIEW

File: code/modules/projectiles/projectile/special.dm [hunk 1]
```diff
@@ -37,7 +36,6 @@
 	desc = "USE A WEEL GUN"
```
Comment (Iamgoofball) [inline code/modules/projectiles/projectile/special.dm]: oh my god

Comment (Chocobro) [inline code/modules/projectiles/projectile/special.dm]: only six years old and he's already talkin' about DRIVE-BYS
